test/helper.sh

```diff
@@ -0,0 +1,216 @@
+# Wait for the given ingress to be fully provisioned.
+# Globals:
+#   None
+# Argument:
+#   Name of the ingress.
+#   Namespace of the ingress.
+# Outputs:
+#   Writes the ingress IP to stdout.
+# Returns:
+#   0 if the ingress IP is populated within 15 minutes, 1 if not.
+###############################################################################
+wait_for_ingress_ip() {
+    local ing ns
+    ing="$1"
+    ns="$2"
+
+    for ATTEMPT in $(seq 30); do
+        vip=$(kubectl get ingress ${ing} -n ${ns} -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+        if [[ ! -z "$vip" ]]; then
+            echo "${vip}"
+            return
+        fi
+        sleep 30
+    done
+    exit 1
+}
+
+###############################################################################
+# Wait for the given Managed Certificate to be fully provisioned.
+# Globals:
+#   None
+# Arguments:
+#   Name of the Managed Certificate.
+#   Namespace of the Managed Certificate.
+# Returns:
+#   0 if the status of the given certificate is Active within 45 minutes, 1 if not.
+###############################################################################
+wait_for_managed_cert() {
+    local mc ns
+    mc="$1"
+    ns="$2"
+
+    for ATTEMPT in $(seq 45); do
+        # Check status of the certificate.
+        certStatus=$(kubectl get ManagedCertificate ${mc} -n ${ns} -o jsonpath="{.status.certificateStatus}")
+        if [[ "${certStatus}" != "Active" ]]; then
+            sleep 60
+            continue
+        fi
+
+        # Check status of each domain.
+        domainStatus=( $(kubectl get ManagedCertificate ${mc} -n ${ns} -o jsonpath="{.status.domainStatus}" | jq  ".[] | .status") )
+        count=${#domainStatus[@]}
+        match=0
+        for status in "${domainStatus[@]}"; do
+            if [[ "${status}" == '"Active"' ]]; then
+                ((match++))
+            fi
+        done
+
+        if [[ "${match}" -eq "${count}" ]]; then
+            return
+        fi
+        sleep 60
+    done
+    exit 1
+}
+
+###############################################################################
+# Basic setup that works for most recipe tests.
+# Globals:
+#   ZONE
+#   SUBNET_REGION
+# Arguments:
+#     Name of the test. Used to generate suffix for resources.
+###############################################################################
+setup_gke_basic() {
+    local test_name
+    test_name="$1"
+    resource_suffix=$(get_hash "${test_name}")
+    resource_name="gke-net-recipes-${resource_suffix}"
+
+    gcloud compute networks create "${resource_name}" --subnet-mode="custom"
+    gcloud compute networks subnets create "${resource_name}" --network="${resource_name}" --region="${SUBNET_REGION}" --range="10.1.2.0/24"
+    gcloud compute instances create "${resource_name}" --zone="${ZONE}" --network="${resource_name}" --subnet="${resource_name}" --image-family="debian-11" --image-project="debian-cloud" --tags="allow-ssh"
```

Comment:
My preference for long command lines is one flag/value pair per line to make this more readable:

```
$ gcloud compute instances create "${resource_name}" \
   --zone="${ZONE}" \
   --network="${resource_name}" \
   --subnet="${resource_name}" \
   --image-family="debian-11" \
   --image-project="debian-cloud" \
   --tags="allow-ssh"
```